conans/test/functional/layout/local_commands_test.py

```diff
@@ -0,0 +1,224 @@
+        assert "Install folder: {}".format(install_folder) in client.out
+
+    def test_local_build_change_base(self):
+        """If we configure a build folder in the layout, the build files in a "conan build ."
+        go to the specified folder: "my_build under the modified base one "common"
+        """
+        # FIXME: The configure is not valid to change the layout, we need the settings and options
+        #        ready
+        client = TestClient()
+        conan_file = str(GenConanfile().with_import("from conans import tools"))
+        conan_file += """
+    def configure(self):
+        self.layout.install.folder = "my_install"
+        self.layout.build.folder = "my_build"
+    def build(self):
+        tools.save("build_file.dll", "bar")
+    """
+        client.save({"conanfile.py": conan_file})
+        client.run("install . -if=common")
+        client.run("build . -if=common/my_install -bf=common")
+        dll = os.path.join(client.current_folder, "common", "my_build", "build_file.dll")
+        assert os.path.exists(dll)
+
+
+class LocalSourceWithLayoutTest(unittest.TestCase):
+
+    def test_local_source(self):
+        """If we configure a source folder in the layout, the downloaded files in a "conan source ."
+        go to the specified folder: "my_source"
+        """
+        # FIXME: The configure is not valid to change the layout, we need the settings and options
+        #        ready
+        client = TestClient()
+        conan_file = str(GenConanfile().with_import("from conans import tools"))
+        conan_file += """
+    def configure(self):
+        self.layout.source.folder = "my_source"
+
+    def source(self):
+        tools.save("downloaded.h", "bar")
+    """
+        client.save({"conanfile.py": conan_file})
+        client.run("install . -if=my_install")
+        # FIXME: This should change to "source ." when "conan source" computes the graph
+        client.run("source . -if=my_install")
+        header = os.path.join(client.current_folder, "my_source", "downloaded.h")
+        assert os.path.exists(header)
+
+    def test_local_source_change_base(self):
+        """If we configure a source folder in the layout, the souce files in a "conan source ."
+        go to the specified folder: "my_source under the modified base one "all_source"
+        """
+        # FIXME: The configure is not valid to change the layout, we need the settings and options
+        #        ready
+        client = TestClient()
+        conan_file = str(GenConanfile().with_import("from conans import tools"))
+        conan_file += """
+    def configure(self):
+        self.layout.install.folder = "my_install"
+        self.layout.source.folder = "my_source"
+
+    def source(self):
+        tools.save("downloaded.h", "bar")
+    """
+        client.save({"conanfile.py": conan_file})
+        client.run("install . -if=common")
+        client.run("source . -if=common/my_install -sf=common")
```

Comment:
Don't we want to deprecate ``-sf``? Maybe it could raise if we are using layouts?

(same for ``-bf``, etc)

Reply:
Yes, I would like to deprecate them because they won't be necessary but until then the "sense" is the "base" folder.

Probably not easy to raise because of two reasons:
- Not sure if when you have available the conanfile (and you can check if you have defined the layout) you lost trace if the directory came from the argument `-sf` etc
- If, for example, we decide that the default "layout.package.folder" in the layout is "package", then is not that easy to check if the value has been modified or not, so it could be tricky.

I would prefer to just introduce the deprecation mechanism at the `conan_api` level when the arguments are declared but in the meantime that is the behavior, they should have.